### PR TITLE
Maya: bug fix the invalid renderlayer name errors out in collect inputs

### PR DESCRIPTION
**Pull request by moonyuet**
## Changelog Description
Collect inputs errors out during maya deadline submission as it can't get the name of renderlayer.
This PR is to help to find the name of the renderlayer.

## Additional info
Links to https://github.com/ynput/ayon-core/pull/173
If you want to test the code, you need to somehow resolve the code above first.
## Testing notes:
1. Create Render Instance in Maya
2. Publish

### CODE REVIEW

File: client/ayon_core/hosts/maya/plugins/publish/collect_inputs.py [hunk 1]
```diff
@@ -180,6 +180,11 @@ def _collect_renderlayer_inputs(self, scene_containers, instance):
             return copy.deepcopy(scene_containers)
         else:
             # Get the members of the layer
+            renderlayer = next((i for i in cmds.ls(type='renderLayer')
+                             if i.endswith(renderlayer)), None)
+            if renderlayer is None:
+                return copy.deepcopy(scene_containers)
```
[the remark this answers is not in the document]
Comment (moonyuet) [reply client/ayon_core/hosts/maya/plugins/publish/collect_inputs.py:186]: > Sorry, what is this doing? And why?
> 
> This is _not_ the correct fix.
> 
> There may be a `BG_main` layer and a `main` layer - one name will still end with the other.
> 
> Why does this need fixing? And what are we trying to solve?

the issue is it errors out as there is not `Main` in renderlayer, but when you check on `cmds.ls(type=renderlayer)` , it has `rs_Main` and `masterlayer`.

[the remark this answers is not in the document]
Comment (BigRoy) [reply client/ayon_core/hosts/maya/plugins/publish/collect_inputs.py:186]: I think the fix is to change `renderlayer = instance.data.get("renderlayer")` to `renderlayer = instance.data.get("setMembers")` - that's at least what we have done. 

One refers to the Maya node for the (legacy) renderlayer - and the other refers to the name of the render setup layer.

Render setup layers always still 'wrap' the legacy renderlayers - so for any rendersetup layer there is a legacy renderlayer - usually with a subtly different name. I believe `setMembers` is that other name.

**Review by BigRoy — APPROVED** [review]
I think this may be the good fix - I'm not sure if we have different changes on the collectors on our end. This would be one to test by @LiborBatek at this stage.

**Review by tokejepsen — APPROVED** [review]
Tested in Maya 2023 succesfully.

**Pull request merged**